database/sqlserver/sqlserver.go

```diff
@@ -181,7 +184,7 @@ func (ss *SQLServer) Lock() error {
 
 // Unlock froms the migration lock from the database
 func (ss *SQLServer) Unlock() error {
-	if !ss.isLocked {
+	if !ss.isLocked.Load() {
```

Comment:
Same here about fixing the local lock optimization implementation

Reply:
thnx,fixed it